google/cloud/storage/internal/rest/stub_test.cc

```diff
@@ -193,6 +194,42 @@ TEST(RestStubTest, CreateBucket) {
               StatusIs(PermanentError().code(), PermanentError().message()));
 }
 
+TEST(RestStubTest, ListBucketsIncludesPageTokenWhenPresentInRequest) {
+  auto mock = std::make_shared<MockRestClient>();
+  std::string expected_token = "test-page-token";
+  ListBucketsRequest request("test-project-id");
+  request.set_page_token(expected_token);
+
+  EXPECT_CALL(*mock,
+              Get(ExpectedContext(),
+                  ResultOf(
+                      "request parameters contain expected pageToken",
+                      [](RestRequest const& r) { return r.parameters(); },
+                      Contains(Pair("pageToken", expected_token)))))
+      .WillOnce(Return(PermanentError()));
+
+  auto tested = std::make_unique<RestStub>(Options{}, mock, mock);
+  auto context = TestContext();
+  tested->ListBuckets(context, TestOptions(), request);
+}
+
+TEST(RestStubTest, ListBucketsOmitsPageTokenWhenEmptyInRequest) {
+  auto mock = std::make_shared<MockRestClient>();
+  ListBucketsRequest request("test-project-id");
+
+  EXPECT_CALL(*mock,
+              Get(ExpectedContext(),
+                  ResultOf(
+                      "request parameters do not contain 'pageToken'",
```

Comment:
It's a small point, but you might also ask yourself why there are gratuitous differences between the description strings of the two matchers:
```
    "request parameters contain expected pageToken"
```
and
```
    "request parameters do not contain 'pageToken'"
```
(one says _"expected"_ and the other quotes `pageToken`).

Consistency is good, and particularly so in tests where matching code patterns help in conveying correctness.

So, I'd suggest changing the first description to ...
```
    "request parameters contain 'pageToken'"
```